scene/2d/animated_sprite_2d.cpp

```diff
@@ -391,7 +397,11 @@ bool AnimatedSprite2D::is_playing() const {
 }
 
 void AnimatedSprite2D::play(const StringName &p_animation, const bool p_backwards) {
-	backwards = p_backwards;
+	if (backwards != p_backwards) {
+		if ((p_backwards && speed_scale > 0.0f) || (!p_backwards && speed_scale < 0.0f)) {
+			set_speed_scale(speed_scale * -1);
+		}
+	}
```

Comment:
Considering the case where `speed_scale` is changed after `play()`, shouldn't this part be handled dynamically, not just at `play()`?

Reply:
As in, every `set_speed_scale()` call after `play()` will automatically flip the sign until the next idle frame?

Reply:
If you want to change the `speed_scale` implicitly by play(), I don't recommend it. The user-set value should be kept the original value, and such inversions should only be done internally.

Reply:
I was trying to to infer your suggestion. You would like if there was a second `animation_direction` variable, which is 1 or -1, used along with `speed_scale`, that only switches sign in `play()`, thus making `speed_scale `**not** change sign?

Reply:
It... would... but I don't know if I fully agree, practically. Setting `speed_scale`  to a negative value, then reversing the direction with `play()`, would make the current animation go forward. It's a double negative. I gather the suggested behaviour is based on **AnimationPlayer**, but I'm struggling to explain the concept.

Reply:
It is theoretically correct that a negative scale play_backward is forward, and I think it makes more sense than implicitly changing speed_scale.

Reply:
I can agree to an extent. I will make the change here, perhaps recycling the current behaviour in another PR if there's a bunch more discussion to be had about specifically that.